src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncMainTests.cs

```diff
@@ -2063,5 +2063,397 @@ static async MyTask Main()
+                        public void Start<TStateMachine>(ref TStateMachine stateMachine) where TStateMachine : IAsyncStateMachine { }
+                        public void SetStateMachine(IAsyncStateMachine stateMachine) { }
+                        public void AwaitOnCompleted<TAwaiter, TStateMachine>(ref TAwaiter awaiter, ref TStateMachine stateMachine) 
+                            where TAwaiter : INotifyCompletion where TStateMachine : IAsyncStateMachine { }
+                        public void AwaitUnsafeOnCompleted<TAwaiter, TStateMachine>(ref TAwaiter awaiter, ref TStateMachine stateMachine)
+                            where TAwaiter : INotifyCompletion where TStateMachine : IAsyncStateMachine { }
+                        public void SetResult(T result) { }
+                        public void SetException(Exception exception) { }
+                        public Threading.Tasks.Task<T> Task => default;
+                    }
+
+                    public static class AsyncHelpers
+                    {
+                        public static void HandleAsyncEntryPoint(Threading.Tasks.Task task)
+                        {
+                            task.GetAwaiter().GetResult();
+                        }
+
+                        public static int HandleAsyncEntryPoint(Threading.Tasks.Task<int> task)
+                        {
+                            return task.GetAwaiter().GetResult();
+                        }
+                    }
+
+                    public struct TaskAwaiter<TResult> : ICriticalNotifyCompletion
+                    {
+                        public bool IsCompleted => true;
+                        public TResult GetResult() => default;
+                        public void OnCompleted(Action continuation) { }
+                        public void UnsafeOnCompleted(Action continuation) { }
+                    }
+
+                    public struct TaskAwaiter : ICriticalNotifyCompletion
+                    {
+                        public bool IsCompleted => true;
+                        public void GetResult() { }
+                        public void OnCompleted(Action continuation) { }
+                        public void UnsafeOnCompleted(Action continuation) { }
+                    }
+                }
+
+                public enum AttributeTargets { All = ~0 }
+
+                [AttributeUsage(AttributeTargets.All)]
+                public sealed class AttributeUsageAttribute : Attribute
+                {
+                    public AttributeUsageAttribute(AttributeTargets validOn) { }
+                    public bool AllowMultiple { get; set; }
+                    public bool Inherited { get; set; }
+                }
+            }
+
+            namespace System.Diagnostics
+            {
+                public class DebuggerStepThroughAttribute : Attribute { }
+            }
+            """;
+
+        [Fact]
```

Comment:
Consider adding WorkItem attributes (unless we only add those for bugs, not features?)